bin/clang/clangGlobals.h

```diff
@@ -83,38 +83,57 @@ class FirstPassASTVisitor;
 struct SSTPragma;
 class SSTReplacePragma;
 class SSTNullVariablePragma;
-struct PragmaConfig {
-  std::map<clang::Decl*,SSTNullVariablePragma*> nullVariables;
-  std::map<clang::FunctionDecl*,std::set<SSTPragma*>> functionPragmas;
-  std::map<const clang::DeclContext*,SSTNullVariablePragma*> nullSafeFunctions;
-  std::vector<std::pair<SSTPragma*, std::string>> globalCppFunctionsToWrite;
-  std::set<std::string> newParams;
-  std::string dependentScopeGlobal;
-  std::string computeMemorySpec;
-  std::list<clang::FunctionDecl*> fxnContexts;
-  std::map<clang::IfStmt*,std::string> predicatedBlocks;
-  std::list<clang::CompoundStmt*> stmtBlocks;
-  union {
-    SkeletonASTVisitor* skeleton;
-    FirstPassASTVisitor* firstPass;
-  } visitor;
 
+/**
+ * @brief The ASTContextLists struct
+ * This contains data structures that keep track of the enclosing context
+ * of the currently being visited statement
+ */
+struct ASTContextLists {
+  std::list<clang::FunctionDecl*> enclosingFunctionDecls;
+  std::list<clang::CompoundStmt*> compoundStmtBlocks;
   clang::CompoundStmt::body_iterator currentStmtBlockBegin(){
-    return stmtBlocks.back()->body_begin();
+    return compoundStmtBlocks.back()->body_begin();
   }
 
   clang::CompoundStmt::body_iterator findStmtBlockMatch(clang::Stmt* s){
-    clang::CompoundStmt* block = stmtBlocks.back();
+    clang::CompoundStmt* block = compoundStmtBlocks.back();
     for (auto* iter = block->body_begin(); iter != block->body_end(); ++iter){
       if (*iter == s) return iter;
     }
     return block->body_end();
   }
 
   clang::CompoundStmt::body_iterator currentStmtBlockEnd(){
-    return stmtBlocks.back()->body_end();
+    return compoundStmtBlocks.back()->body_end();
   }
+};
 
+/**
+ * @brief The ASTMarkings struct
```

Comment:
Maybe we can call this ASTNodeMetaData or something, just a different name than Markings?

Reply:
Changed.